From: =?utf-8?q?Piotr_O=C5=BCarowski?= <piotr@debian.org>
Date: Tue, 12 May 2020 12:32:01 +0200
Subject: calibre: ebook-convert to pdf aborts due to missing objects in
 QWebEngineProfile
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Forwarded: not-needed
Closes: #944875
Thanks: Piotr Ożarowski <piotr@debian.org>

commit 6d030410a363123c5152d4b2d1a056bac8bffa46 (HEAD -> master)
Author: Piotr Ożarowski <piotr@debian.org>
Date:   Tue May 12 12:32:01 2020 +0200

    use deprecated setRequestInterceptor if setUrlRequestInterceptor is not available
---
 src/calibre/ebooks/pdf/html_writer.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/calibre/ebooks/pdf/html_writer.py b/src/calibre/ebooks/pdf/html_writer.py
index d3bd5b0..082b117 100644
--- a/src/calibre/ebooks/pdf/html_writer.py
+++ b/src/calibre/ebooks/pdf/html_writer.py
@@ -275,7 +275,10 @@ class RenderManager(QObject):
         ans.setHttpUserAgent(ua)
         s = ans.settings()
         s.setDefaultTextEncoding('utf-8')
-        ans.setUrlRequestInterceptor(self.interceptor)
+        try:
+            ans.setUrlRequestInterceptor(self.interceptor)
+        except AttributeError:
+            ans.setRequestInterceptor(self.interceptor)
         self.profile = ans
 
         self.opts = opts
